shopinvader_anonymous_partner/models/res_partner.py

```diff
@@ -101,3 +102,26 @@ def _get_anonymous_partner__cookie(self, cookies: Cookies):
         if not token:
             return self.env["res.partner"].sudo().browse()
         return self._get_anonymous_partner__token(token)
+
+    def _promote_from_anonymous_partner(self, anonymous_partner: ResPartner):
+        """
+        Promote an anonymous partner to an authenticated partner.
+
+        This method should be overridden by other modules to implement
+        the partner resolution logic, merging the anonymous partner cart
+        for instance.
+        """
+
+    @api.model
+    def _promote_anonymous_partner(
+        self, partner: ResPartner, cookies: Cookies, response: Response
+    ):
+        """
+        Promote the current anonymous partner to the given authenticated partner.
+
+        This method calls the partner promotion and removes the anonymous partner cookie.
+        """
+        anonymous_partner = self._get_anonymous_partner__cookie(cookies)
+        if anonymous_partner:
+            partner._promote_from_anonymous_partner(anonymous_partner)
+            self._delete_anonymous_partner__cookie(cookies, response)
```

Comment:
I think I would leave this logic in `shopinvader_api_signin_jwt` because it is tricky and so closely related to the authentication method.

Reply:
Maybe, but I don't think we should encourage re-use or override of that part. It is only 4 lines and quite delicate, so I feel it's better to copy it (probably with subtle changes) in other signin methods.

Reply:
That's why the `_promote_from_anonymous_partner` method exists, that's the one that should be inherited to implement a promotion mechanism.

Reply:
> That's why the `_promote_from_anonymous_partner` method exists, that's the one that should be inherited to implement a promotion mechanism.

Yes that part is ok. It's specifically this `_promote_anonymous_partner` method that I'm not comfortable with.

Reply:
Yes I know, I just don't understand why :). If you override the `_promote_anonymous_partner` in the `shopinvader_anonymous_partner` you can only mean to change the promotion mechanism of anonymous partners and in this case yeah you can break the anonymous partner flow. 

I am certainly missing something here, but I expect a `shopinvader_anonymous_partner` module to handle the flow of anonymous partners regardless of how the identified partner authentication is implemented.

Reply:
> just don't understand why

It makes the logic of the signin route much harder to follow, for no real benefit.